### PR TITLE
FALCON-2207 ACL check while deleting extensions

### PR DESCRIPTION
**Pull request by PracheerAgarwal**

### CODE REVIEW

File: extensions/src/test/java/org/apache/falcon/extensions/store/ExtensionStoreTest.java [hunk 1]
```diff
@@ -25,6 +25,7 @@
 import org.apache.falcon.extensions.jdbc.ExtensionMetaStore;
 import org.apache.falcon.extensions.mirroring.hdfs.HdfsMirroringExtension;
 import org.apache.falcon.hadoop.JailedFileSystem;
+import org.apache.falcon.security.CurrentUser;
```
Comment (PraveenAdlakha) [inline extensions/src/test/java/org/apache/falcon/extensions/store/ExtensionStoreTest.java]: seem unnecessary, we are not using it in this class checkstyle might fail.

Comment (PracheerAgarwal) [inline extensions/src/test/java/org/apache/falcon/extensions/store/ExtensionStoreTest.java]: ack.

File: extensions/src/test/java/org/apache/falcon/extensions/jdbc/ExtensionMetaStoreTest.java [hunk 2]
```diff
@@ -60,7 +60,7 @@ public void init() {
     @Test
     public void testExtension(){
         //insert
-        stateStore.storeExtensionBean("test1", "test_location", ExtensionType.TRUSTED, "test_description");
+        stateStore.storeExtensionBean("test1", "test_location", ExtensionType.TRUSTED, "test_description", "falcon");
```
Comment (sandeepSamudrala) [inline extensions/src/test/java/org/apache/falcon/extensions/jdbc/ExtensionMetaStoreTest.java]: rename to falconUser just for easy readability.

Comment (sandeepSamudrala) [inline extensions/src/test/java/org/apache/falcon/extensions/jdbc/ExtensionMetaStoreTest.java]: Remaining places too in this class.

File: extensions/src/test/java/org/apache/falcon/extensions/store/ExtensionStoreTest.java [hunk 3]
```diff
@@ -104,32 +104,57 @@ public void clean() {
         }
     }
 
-
     @Test
     public void testRegisterExtension() throws IOException, URISyntaxException, FalconException{
-        createLibs();
-        createReadmeAndJar();
-        createMETA();
+        String extensionPath = EXTENSION_PATH + "testRegister";
```
Comment (sandeepSamudrala) [inline extensions/src/test/java/org/apache/falcon/extensions/store/ExtensionStoreTest.java]: Can you add code to clean up all the fs directories that are getting created at the end of all the tests. You can clean up the root directory EXTENSION_PATH directly.

**Comment by pallavi-rao** [comment]
👍 . Will merge once @sandeepSamudrala's comments are addressed.

**Comment by sandeepSamudrala** [comment]
Looks Good :+1:

**Pull request closed without merging**